hera_mc/weather.py

```diff
@@ -57,6 +57,8 @@ def _reduce_time_vals(times, vals, period, strategy='decimate'):
 
     # the // operator is a floored divide.
     times_keep, inds = np.unique((times // period) * period, return_index=True)
+    times_keep = sorted(times_keep)
+    inds = sorted(inds)
```

Comment:
I am quite sure this is not what we want; since each array gets sorted on its own terms, the matching of times and indices will be broken unless both have the same rank-ordering in their respective arrays, i.e. that the times are already sorted. To sort both arrays using the same key, you'd use something like the return value from `np.argsort(times)`. But again, the way that `np.unique` works, everything should Just Work without having to apply any extra sorts before or after. Reading the code, I have trouble seeing how this might break down — the key requirement is the `vals` always be indexed through `inds`, which appears to be the case. Can you come up with a minimal example set of inputs that yields bad outputs?